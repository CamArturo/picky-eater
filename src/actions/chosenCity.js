export const addChosenCity = (city) => {
  return {
    type: 'ADD_CHOSEN_CITY',
    city
  }
};