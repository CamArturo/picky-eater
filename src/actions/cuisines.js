export const sendCuisinesToStore = (cuisines) => {
  return {
    type: 'ADD_AVAILABLE_CUISINES',
    cuisines
  };
};