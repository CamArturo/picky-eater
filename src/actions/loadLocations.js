export const sendLocationsToStore = (locations) => (
  {
    type: 'STORE_CITIES',
    locations
  }
);