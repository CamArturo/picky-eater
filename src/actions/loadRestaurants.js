export const sendRestaurantsToStore = (restaurants) => (
  {
    type: 'STORE_RESTAURANTS',
    restaurants
  }
);