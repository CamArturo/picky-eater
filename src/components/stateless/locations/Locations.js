import React from 'react';

const Locations = () => {
  return (
    <section className="locations-container">
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
      <button>Newton, TX</button>
    </section>
  );
};

export default Locations;
