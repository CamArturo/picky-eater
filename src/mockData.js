export const chosenCity = {
  city: 'Newton, IL',
  id: 5172
};

export const mockLocations = [
  {
    'id': 7171,
    'name': 'Newton Grove, NC',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 95,
    'state_name': 'North Carolina',
    'state_code': 'NC'
  },
  {
    'id': 5172,
    'name': 'Newton, IL',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 82,
    'state_name': 'Illinois',
    'state_code': 'IL'
  },
  {
    'id': 6946,
    'name': 'Newton, MS',
    'country_id': 216,
    'country_name': 'United States',
    'country_flag_url': 'https://b.zmtcdn.com/images/countries/flags/country_216.png',
    'should_experiment_with': 0,
    'discovery_enabled': 0,
    'has_new_ad_format': 0,
    'is_state': 0,
    'state_id': 93,
    'state_name': 'Mississippi',
    'state_code': 'MS'
  }
];

export const availableCuisines = [
  {
    'cuisine': {
      'cuisine_id': 6,
      'cuisine_name': 'Afghani'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 152,
      'cuisine_name': 'African'
    }
  },
  {
    'cuisine': {
      'cuisine_id': 1,
      'cuisine_name': 'American'
    }
  }
];

export const mockRestaurants = {
  restaurants:
    {
      results_found: 7,
      results_start: 0,
      results_shown: 7,
      restaurants:
        [[Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object],
          [Object]]
    }
};

export const mockListRestaurants = [
  'Yesterday\'s Pub & Dining',
  'Olde Tyme Steakhaus',
  'Brew City Pub & Grill',
  'Holiday Restaurant',
  'Richard\'s Farm Restaurant',
  'Lone Star Steakhouse & Saloon',
  'Gray\'s'
];