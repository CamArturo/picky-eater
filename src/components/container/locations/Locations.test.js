// returning an object that has proper object with key and value of a function
// => aka the expected object.

// dispatch is being called with correct action
import { shallow, mount, render } from 'enzyme';
import { Locations } from './Locations';
import React from 'react';
import { mockLocations } from '../../../mockData';
import { mapStateToProps, mapDispatchToProps } from './Locations';
import { addChosenCity } from '../../../actions/chosenCity';

describe('Locations', () => {
  let wrapper;
  let mockState;
  let mockProps;
  beforeEach(() => {
    mockState = {
      locations: mockLocations,
      chosenCity: 'Newton, IL'
    };

    mockProps = {
      storeChosenCity: jest.fn()
    };

    wrapper = shallow(<Locations {...mockProps} locations={mockLocations} storeChosenCity={jest.fn()} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should mapStateToProps', () => {
    const actual = mapStateToProps(mockState);
    const expected = {
      locations: mockLocations
    };
    expect(actual).toEqual(expected);
  });
  it('calls dispatch with an addChosenCity when a button is clicked', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = {'city': 'Newton, IL', 'type': 'ADD_CHOSEN_CITY'};

    const mappedProps = mapDispatchToProps(mockDispatch); // <---- action creator

    mappedProps.storeChosenCity('Newton, IL');
    // dispatches a function that passing that object the actions
    // method that calls another method(action creator) that has an action (an object)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
  it('should fire storeChosenCity when a button is clicked', () => {
    const mockedDispatchToProps = jest.fn();
    const mappedProps = mapDispatchToProps(mockedDispatchToProps);
    const chosenCity = 'Newton, IL';

    mappedProps.storeChosenCity(chosenCity);

    const expected = addChosenCity(chosenCity); // everything that is going to be called with
    expect(mockedDispatchToProps).toHaveBeenCalledWith(expected);

  });
  it('should only render buttons when there are more than 0 locations in store', () => {
    const mockLocations = [];
    const mockedDispatchToProps = jest.fn();
    const mappedProps = mapDispatchToProps(mockedDispatchToProps);
    const wrapper2 = shallow(<Locations storeChosenCity={mappedProps} locations={mockLocations} />);

    expect(wrapper2.find('.location-btn').exists()).toEqual(false);
  });
  it('should not render buttons if locations is zero length', () => {
    const mockState3 = {
      locations: undefined,
      chosenCity: 'Newton, IL'
    };
    const wrapper3 = shallow(<Locations {...mockState3} {...mockProps} locations={mockLocations} storeChosenCity={jest.fn()} />);
    expect(wrapper3).toMatchSnapshot();
  });
});

