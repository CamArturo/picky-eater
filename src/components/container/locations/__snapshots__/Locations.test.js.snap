// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Locations
    locations={
        Array [
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 7171,
              "is_state": 0,
              "name": "Newton Grove, NC",
              "should_experiment_with": 0,
              "state_code": "NC",
              "state_id": 95,
              "state_name": "North Carolina",
            },
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 5172,
              "is_state": 0,
              "name": "Newton, IL",
              "should_experiment_with": 0,
              "state_code": "IL",
              "state_id": 82,
              "state_name": "Illinois",
            },
            Object {
              "country_flag_url": "https://b.zmtcdn.com/images/countries/flags/country_216.png",
              "country_id": 216,
              "country_name": "United States",
              "discovery_enabled": 0,
              "has_new_ad_format": 0,
              "id": 6946,
              "is_state": 0,
              "name": "Newton, MS",
              "should_experiment_with": 0,
              "state_code": "MS",
              "state_id": 93,
              "state_name": "Mississippi",
            },
          ]
    }
    storeChosenCity={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/cuisine"
>
          <button
                    className="location-btn"
                    onClick={[Function]}
          >
                    Newton Grove, NC
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/cuisine"
>
          <button
                    className="location-btn"
                    onClick={[Function]}
          >
                    Newton, IL
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/cuisine"
>
          <button
                    className="location-btn"
                    onClick={[Function]}
          >
                    Newton, MS
          </button>
</NavLink>,
      ],
      "className": "locations-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "nav0",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="location-btn"
            onClick={[Function]}
>
            Newton Grove, NC
</button>,
          "to": "/cuisine",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "key0",
          "nodeType": "host",
          "props": Object {
            "children": "Newton Grove, NC",
            "className": "location-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton Grove, NC",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "nav1",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="location-btn"
            onClick={[Function]}
>
            Newton, IL
</button>,
          "to": "/cuisine",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "key1",
          "nodeType": "host",
          "props": Object {
            "children": "Newton, IL",
            "className": "location-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton, IL",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "nav2",
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button
            className="location-btn"
            onClick={[Function]}
>
            Newton, MS
</button>,
          "to": "/cuisine",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "key2",
          "nodeType": "host",
          "props": Object {
            "children": "Newton, MS",
            "className": "location-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Newton, MS",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/cuisine"
>
            <button
                        className="location-btn"
                        onClick={[Function]}
            >
                        Newton Grove, NC
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/cuisine"
>
            <button
                        className="location-btn"
                        onClick={[Function]}
            >
                        Newton, IL
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/cuisine"
>
            <button
                        className="location-btn"
                        onClick={[Function]}
            >
                        Newton, MS
            </button>
</NavLink>,
        ],
        "className": "locations-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "nav0",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="location-btn"
              onClick={[Function]}
>
              Newton Grove, NC
</button>,
            "to": "/cuisine",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "key0",
            "nodeType": "host",
            "props": Object {
              "children": "Newton Grove, NC",
              "className": "location-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Newton Grove, NC",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "nav1",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="location-btn"
              onClick={[Function]}
>
              Newton, IL
</button>,
            "to": "/cuisine",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "key1",
            "nodeType": "host",
            "props": Object {
              "children": "Newton, IL",
              "className": "location-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Newton, IL",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "nav2",
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button
              className="location-btn"
              onClick={[Function]}
>
              Newton, MS
</button>,
            "to": "/cuisine",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "key2",
            "nodeType": "host",
            "props": Object {
              "children": "Newton, MS",
              "className": "location-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Newton, MS",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
